product_variants_no_automatic_creation/views/product_view.xml

```diff
@@ -17,18 +17,54 @@
             <field name="model">product.template</field>
             <field name="inherit_id" ref="product.product_template_only_form_view" />
             <field name="arch" type="xml">
-                <button string="Variant Prices" position="attributes">
+            <xpath expr="//button[@name='103']" position="attributes">
+               
                     <attribute name="invisible">1</attribute>
-                </button>
-                <button string="Variant Prices" position="after">
+               </xpath>
+            <xpath expr="//button[@name='103']" position="after">   
+               
                     <button class="oe_inline oe_stat_button" name="action_open_attribute_prices"
                         icon="fa-strikethrough" type="object" string="Variant Prices">
                     </button>
-                </button>
-                <field name="attribute_line_ids" position="before">
+                </xpath>
+                <xpath expr="//field[@name='attribute_line_ids']" position="before">   
+               
                     <group>
                         <field name="no_create_variants" />
                     </group>
+                </xpath>
+            </field>
+        </record>
+
+        <record model="ir.ui.view" id="product_variants_view">
```

Comment:
This is copied-pasted from 8.0, but you have to update all the changes introduce in 8.0. See https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0